src/quantile.js

```diff
@@ -1,8 +1,23 @@
-import ascending from "./ascending.js";
+import max from "./max.js";
+import min from "./min.js";
+import quickselect from "./quickselect.js";
 import number, {numbers} from "./number.js";
 
-export default function quantile(values, p, valueof) {
-  return quantileSorted(Float64Array.from(numbers(values, valueof)).sort(ascending), p);
+export default function quantile(values, p, valueof = number) {
+  var v = Float64Array.from(numbers(values, valueof));
+  if (!(n = v.length)) return;
+  if ((p = +p) <= 0 || n < 2) return min(v);
+  if (p >= 1) return max(v);
+  var n,
+      i = (n - 1) * p,
+      i0 = Math.floor(i);
+  quickselect(v, i0);
+  var value0 = -Infinity, value1 = +Infinity;
+  for (var j = 0; j < n; j++) {
+    if (j <= i0 && v[j] > value0) value0 = v[j];
+    if (j > i0 && v[j] < value1) value1 = v[j];
+  }
```

Comment:
What about `min(v.subarray(0, i0))` and `max(v.subarray(i0, n))` or similar?

Reply:
:blush: